src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -143,12 +142,14 @@ public Drivetrain(ShuffleboardTab drivetrainTab, ShuffleboardTab swerveModulesTa
 
     m_poseEstimator = new SwerveDrivePoseEstimator(
       DriveConstants.kKinematics,
-      m_pigeon.getRotation2d(),
+      getYaw(),
       getModulePositions(),
       new Pose2d() // initial Odometry Location
     );
     m_poseEstimator.setVisionMeasurementStdDevs(VisionConstants.kBaseVisionPoseStdDevs);
 
+    setPigeonYaw(DriveConstants.kStartingHeadingDegrees);
+
```

Comment:
why dose the PigeonYaw get reset after initalizeing the pose estimator

Reply:
it will reset the odometry as well